### DISCUSSION TITLE
IKEv2 package loss in network

### DISCUSSION THREAD

**Discussion by keskinadem** [strongswan/strongswan#2314] [General]
Post Quantum Key exchanges need more IKE packages in network. The tunnel which could be established in 4 IKE messages now may cost dozens of IKE messgaes.(including framgmented IKE messages) What if one of IKE fragment message can not be delivered to the receiver? Is there any solution to this problem?

**Comment by tobiasbrunner** [comment]
IKE messages are retransmitted (see the [related settings](https://docs.strongswan.org/docs/5.9/config/retransmission.html)), whether they are fragmented or not.
